wsdl_docs.inc

```diff
@@ -606,32 +617,63 @@ function _wsdl_docs_service_import($name) {
   else {
     return "No service found with this name.";
   }
-  // Import via file.
-  // Manipulate $_FILES to match form version.
-  $_FILES['files'] = $_FILES['soap_api_definition'];
-  // Confirm file is of correct type.
-  $tmp = explode(".", $_FILES['soap_api_definition']['name']);
-  $ext = array_pop($tmp);
-  if ($ext != 'wsdl') {
-    return services_error(t('incorrect file format'), 500);
+  // Create $_FILES['files'] from array of file information provided by REST request.
+  $_FILES['files'] = array();
+  foreach ($_FILES['soap_api_definition'] as $file_key => $values) {
+    foreach ($values as $key => $value) {
+      $_FILES['files'][$key][$file_key] = $value;
+    }
+  }
+
+  // First, confirm file(s) of correct extension.
+  foreach ($_FILES['files'] as $key => $file) {
```

Comment:
You can use `https://secure.php.net/pathinfo` here to get the extension directly.

Reply:
Thanks - fixed it (in 2 places)

Reply:
You can use `pathinfo($file['name'], PATHINFO_EXTENSION)` to get just the extension

Reply:
Thanks - missed that.